tools/prepare_software_list.py

```diff
@@ -0,0 +1,290 @@
+'''
+Download / prepare / process XMPP DOAP files for the software list
+Requires: Pillow, python-slugify
+'''
+from typing import Optional
+from typing import Union
+from typing import cast
+
+from datetime import datetime
+from datetime import timedelta
+from pathlib import Path
+import json
+import os
+import shutil
+import sys
+from urllib.parse import urlparse
+from xml.etree import ElementTree as ET
```

Comment:
https://docs.python.org/3/library/xml.html

> **Warning**
> 
> The XML modules are not secure against erroneous or maliciously constructed data. If you need to parse untrusted or unauthenticated data see the [XML vulnerabilities](https://docs.python.org/3/library/xml.html#xml-vulnerabilities) and [The defusedxml Package](https://docs.python.org/3/library/xml.html#defusedxml-package) sections.

As we are downloading XML from random servers, we have to ensure that it's not going to blow up, especially now that we build the website on our servers again and not in GitHub's or Travis' CI :).

Reply:
switched to `defusedxml` for parsing